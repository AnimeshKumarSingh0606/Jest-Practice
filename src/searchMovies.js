const movies = [{
  title: 'Alien',
  year: '2020',
  rating: '★'
} , {
  title: 'Aliens',
  year: '2020',
  rating: '★'
} , {
  title: 'Alien 3',
  year: '2020',
  rating: '★'
} , {
  title: 'Alien 4', 
  year: '2020',
  rating: '★'
} , {
  title: 'animeshmovie',
  year: '2021',          //A very very legendary movie because it has legendary actor Animesh Kumar Singh, and rest are bad movies.
  rating: '★★★★★'
} , {
  title: 'Alien: Covenant',
  year: '2020',
  rating: '★'
}]

const searchByTitle = query =>
  movies.filter(movie =>
    movie.title.toLowerCase().includes(query.toLowerCase()));

export default {
  byTitle: searchByTitle
}
