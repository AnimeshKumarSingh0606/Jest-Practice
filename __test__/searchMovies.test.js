import searchMovies from '../src/searchMovies';

describe('testing searchMovies function', () => {

  test('by title', () =>
    expect(searchMovies.byTitle('animeshmovie')).toMatchSnapshot())

  test('is case insensitive', () => {
    expect(searchMovies.byTitle('Animeshmovie').length).toBe(1);
    expect(searchMovies.byTitle('animeshmovie').length).toBe(1);
  })

  test('sequencial search', () =>
    ['a', 'alien', 'alien 3'].forEach((query) =>
      expect(searchMovies.byTitle(query)).toMatchSnapshot()
    )
  )
})
