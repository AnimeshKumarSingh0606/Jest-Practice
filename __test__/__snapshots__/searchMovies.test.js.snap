

exports[`testing searchMovies function by title 1`] = `
Array [
  Object {
    "rating": "★★★★★",
    "title": "animeshmovie",
    "year": "2021",
  },
] 
`;

exports[`testing searchMovies function sequencial search 1`] = `
Array [
  Object {
    "rating": "★",
    "title": "Alien",
    "year": "2020",
  },
  Object {
    "rating": "★",
    "title": "Aliens",
    "year": "2020",
  },
  Object {
    "rating": "★",
    "title": "Alien 3",
    "year": "2020",
  },
  Object {
    "rating": "★",
    "title": "Alien 4",
    "year": "2020",
  },
  Object {
    "rating": "★",
    "title": "Alien: Covenant",
    "year": "2020",
  },
]
`;

exports[`testing searchMovies function sequencial search 2`] = `
Array [
  Object {
    "rating": "★",
    "title": "Alien",
    "year": "2020",
  },
  Object {
    "rating": "★",
    "title": "Aliens",
    "year": "2020",
  },
  Object {
    "rating": "★",
    "title": "Alien 3",
    "year": "2020",
  },
  Object {
    "rating": "★",
    "title": "Alien 4",
    "year": "2020",
  },
  Object {
    "rating": "★",
    "title": "Alien: Covenant",
    "year": "2020",
  },
]
`;

exports[`testing searchMovies function sequencial search 3`] = `
Array [
  Object {
    "rating": "★",
    "title": "Alien 3",
    "year": "2020",
  },
]
`;
